src/jit/lsrabuild.cpp

```diff
@@ -0,0 +1,3021 @@
+        RefPosition*   pos        = newRefPosition(srcInterval, currentLoc, RefTypeUse, refPosNode, candidates, 0);
+        if (delayRegFree)
+        {
+            pos->delayRegFree = true;
+        }
+
+        if (useNode->IsRegOptional())
+        {
+            pos->setAllocateIfProfitable(true);
+        }
+        consumed++;
+
+        // Create additional use RefPositions for multi-reg nodes.
+        for (int idx = 1; idx < locInfo.info.dstCount; idx++)
+        {
+            noway_assert(srcInterval->relatedInterval != nullptr);
+            srcInterval = srcInterval->relatedInterval;
+#ifdef _TARGET_ARM_
+            if (useNode->OperIsPutArgSplit() ||
+                (compiler->opts.compUseSoftFP && (useNode->OperIsPutArgReg() || useNode->OperGet() == GT_BITCAST)))
+            {
+                // get first candidate, set bits in useCandidates must be in sequential order.
+                candidates = genFindLowestReg(allCandidates);
+                allCandidates &= ~candidates;
+            }
+#endif // _TARGET_ARM_
+            RefPosition* pos = newRefPosition(srcInterval, currentLoc, RefTypeUse, refPosNode, candidates, idx);
+            consumed++;
+        }
+    }
+
+    assert(consumed == consume);
+    if (consume != 0)
+    {
+        listNodePool.ReturnNodes(useList);
+    }
+
+    buildInternalRegisterUsesForNode(tree, info, internalRefs, internalCount);
+
+    RegisterType registerType  = getDefType(tree);
+    regMaskTP    candidates    = info->getDstCandidates(this);
+    regMaskTP    useCandidates = info->getSrcCandidates(this);
+
+#ifdef DEBUG
+    if (VERBOSE && produce)
+    {
+        printf("Def candidates ");
+        dumpRegMask(candidates);
+        printf(", Use candidates ");
+        dumpRegMask(useCandidates);
+        printf("\n");
+    }
+#endif // DEBUG
+
+#if defined(_TARGET_AMD64_)
+    // Multi-reg call node is the only node that could produce multi-reg value
+    assert(produce <= 1 || (tree->IsMultiRegCall() && produce == MAX_RET_REG_COUNT));
+#endif // _TARGET_xxx_
+
+    // Add kill positions before adding def positions
+    buildKillPositionsForNode(tree, currentLoc + 1);
+
+#if FEATURE_PARTIAL_SIMD_CALLEE_SAVE
+    VARSET_TP liveLargeVectors(VarSetOps::UninitVal());
+    if (enregisterLocalVars && (RBM_FLT_CALLEE_SAVED != RBM_NONE))
+    {
+        // Build RefPositions for saving any live large vectors.
+        // This must be done after the kills, so that we know which large vectors are still live.
+        VarSetOps::AssignNoCopy(compiler, liveLargeVectors, buildUpperVectorSaveRefPositions(tree, currentLoc + 1));
+    }
+#endif // FEATURE_PARTIAL_SIMD_CALLEE_SAVE
+
+    ReturnTypeDesc* retTypeDesc    = nullptr;
+    bool            isMultiRegCall = tree->IsMultiRegCall();
+    if (isMultiRegCall)
+    {
+        retTypeDesc = tree->AsCall()->GetReturnTypeDesc();
+        assert((int)genCountBits(candidate
```

Comment:
Needs a method header comment

Reply:
This and your next several comments aren't placed at a method, so I can't tell what they're referring to.

Reply:
OK, This must be a bug in the review tool for GitHub.
I did click on the correct line in this file when I submitted the comment.
You can use you judgement on all of these.